### PR TITLE
Bug 1500930 - Prevent multiple deprovision pods from spawning if a job is already in progress

### PR DESCRIPTION
**Pull request by eriknelson**
**Describe what this PR does and why we need it**:
Returns the correct 202 Accepted response if a deprovision job is already in progress for a requested ServiceInstance DELETE.

### CODE REVIEW

**Comment by djzager** [comment]
Test steps:
1. Provision postgresql
2. Deprovision postgresql

### Before fix:

```
$ oc get namespaces
NAME                                   STATUS    AGE
...
19c38d71-9ee2-4dc8-ac2e-cdbeb12b8f32   Active    7s
3b7fe989-92e9-458b-86de-96524df6e054   Active    7s
657a984e-e3e0-41d2-b22f-de465dbdea3a   Active    7s
ansible-service-broker                 Active    5m
b2a4833f-8dc8-4811-ad69-bd17ac14b123   Active    7s
d7d4d2ba-faf4-4038-8464-488332f0c353   Active    7s
...
```

### After fix:

```
$ oc get namespaces
NAME                                   STATUS    AGE
ansible-service-broker                 Active    9m
bab62a51-b76a-4a20-aa2d-098aecf682bb   Active    2s
bar-project                            Active    51s
default                                Active    11m
foo-project                            Active    5m
kube-public                            Active    11m
kube-service-catalog                   Active    11m
kube-system                            Active    11m
myproject                              Active    10m
openshift                              Active    11m
openshift-infra                        Active    11m
openshift-template-service-broker      Active    10m
```

File: pkg/apb/types.go [hunk 1]
```diff
@@ -88,11 +88,29 @@ type ExtractedCredentials struct {
...
+
+	// JobStateAPBMethodTypeDeprovision - Deprovision MethodType const.
+	JobStateAPBMethodTypeDeprovision JobStateAPBMethodType = "deprovision"
+
+	// JobStateAPBMethodTypeBind - Bind MethodType const.
+	JobStateAPBMethodTypeBind JobStateAPBMethodType = "bind"
+
+	// JobStateAPBMethodTypeUnbind - Unbind MethodType const.
+	JobStateAPBMethodTypeUnbind JobStateAPBMethodType = "unbind"
+)
```
Comment (jmrodri) [inline pkg/apb/types.go]: Could you make the variable type a little longer? OOF! add this to tech debt :)

**Comment by eriknelson** [comment]
Thank you for the test @djzager !

File: pkg/apb/types.go (hunk 1 above)
Comment (jmrodri) [reply pkg/apb/types.go]: `JobMethodProvision JobMethod = "provision"`

**Comment by jmrodri** [comment]
Merging as others have tested this.

**Comment by rthallisey** [comment]
https://github.com/openshift/ansible-service-broker/issues/387

**Pull request merged**